src/shogun/optimization/FirstOrderPenalizedStochasticCostFunction.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) The Shogun Machine Learning Toolbox
+ * Written (w) 2015 Wu Lin
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+
+#ifndef FIRSTORDERPENALIZEDSTOCHASTICCOSTFUNCTION_H
+#define FIRSTORDERPENALIZEDSTOCHASTICCOSTFUNCTION_H
+#include <shogun/lib/config.h>
+#include <shogun/optimization/FirstOrderStochasticCostFunction.h>
+#include <shogun/optimization/FirstOrderPenalizedCostFunction.h>
+namespace shogun
+{
+/** @brief The first order penalized stochastic cost function base class.
+ *
+ * note that the cost function must be the finite sum of sample-specific cost
```

Comment:
this is class for  sample-specific cost functions
the type of cost functions can be written as
f(w)= \sum_i{f_i(w)}
where i is the index of samples

For example,
lease squares loss
f(w)=0.5 * \sum_i{ (y_i-w'*x_i)^2 },
where (x_i,y_i) are features and a label  for the i-th sample.